File: Python/program/Motorcycle.py
# Saya Boy Aditya Rohmaulana NIM 2203488 mengerjakan
# soal Latihan Praktikum 4 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin

from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __jenis_motor = ""
    __kapasitas_tangki = 0
    
    def __init__(self, plat_nomor = "", merk = "", tahun_produksi = "", warna = "", jenis_motor = "", kapasitas_tangki = ""):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)
        self.__jenis_motor = jenis_motor
        self.__kapasitas_tangki = kapasitas_tangki

    def get_jenis_motor(self):
        return self.__jenis_motor

    def set_jenis_motor(self, jenis_motor):
        self.__jenis_motor = jenis_motor

    def get_kapasitas_tangki(self):
        return self.__kapasitas_tangki

    def set_kapasitas_tangki(self, kapasitas_tangki):
        self.__kapasitas_tangki = kapasitas_tangki

File: CPP/program/Motorcycle.cpp
/*Saya Boy Aditya Rohmaulana NIM 2203488 mengerjakan
soal Latihan Praktikum 4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#pragma once

#include "Vehicle.cpp"
#include <string>
using namespace std;

class Motorcycle : public Vehicle {
  private:
    string jenis_motor;
    int kapasitas_tangki;

  public:
    //   Constructor
    Motorcycle() {
    }

    Motorcycle(string jenis_motor, string plat_nomor, string merk, int tahun_produksi, string warna) : Vehicle(plat_nomor, merk, tahun_produksi, warna) {
        this->jenis_motor = jenis_motor;
        kapasitas_tangki = -1;
    }

    Motorcycle(string jenis_motor, int kapasitas_tangki, string plat_nomor, string merk, int tahun_produksi, string warna) : Vehicle(plat_nomor, merk, tahun_produksi, warna) {
        this->jenis_motor = jenis_motor;
        this->kapasitas_tangki = kapasitas_tangki;
    }

    // Getter and Setter
    string getJenisMotor() {
        return jenis_motor;
    }

    void setJenisMotor(string jenis_motor) {
        this->jenis_motor = jenis_motor;
    }

    int getKapasitasTangki() {
        return kapasitas_tangki;
    }

    void setKapasitasTangki(int kapasitas_tangki) {
        this->kapasitas_tangki = kapasitas_tangki;
    }

    // Destructor
    ~Motorcycle() {
    }
};